static/src/javascripts/lib/ajax.js

```diff
@@ -26,6 +25,8 @@ const ajax = (params) => {
         credentials: options.withCredentials ? 'include' : undefined,
     }
 
+    if (['GET', 'HEAD'].includes(`${method}`.toUpperCase())) delete init.body;
+
```

Comment:
This ensures no “empty object” gets passed on `GET` or `HEAD` requests.

Reply:
Might be worth adding this as a comment to the code so that it's clearer to anyone who reads it.
```suggestion
    // Ensure no “empty object” gets passed on GET or HEAD requests.
    if (['GET', 'HEAD'].includes(`${method}`.toUpperCase())) delete init.body;

```

Reply:
Yes, to be honest, I want to get rid of `ajax.js` altogether. See #23624